app/services/authorizer.rb

```diff
@@ -33,20 +33,43 @@ def find_collection(resource_class, options = {})
                                                           *values]).uniq
 
     all_filters = all_filters.all # load all records, so #empty? does not call extra COUNT(*) query
-    return resource_class.where('1=0') if all_filters.empty?
 
-    unless @base_collection.nil?
-      if @base_collection.empty?
-        return resource_class.where('1=0')
-      else
-        resource_class = resource_class.where(:id => base_ids)
+    scope_components = build_filtered_scope_components(resource_class, all_filters, options)
```

Comment:
This whole method seems a bit hard to read, useless comment though, I don't know how to make it much easier to read :disappointed: